app/components/file-renderer/component.ts

```diff
@@ -0,0 +1,42 @@
+import { action, computed } from '@ember-decorators/object';
+import Component from '@ember/component';
+import config from 'ember-get-config';
+
+/**
+ * @module ember-osf
+ * @submodule components
+ */
+
+/**
+ * Render the provided url in an iframe via MFR
+ *
+ * Sample usage:
+ * ```handlebars
+ * {{file-renderer
+ *   download=model.links.download
+ *     width="800" height="1000" allowfullscreen=true}}
+ * ```
+ * @class FileRenderer
+ */
+export default class FileRenderer extends Component {
+    download: string;
+    width: string = this.width || '100%';
+    height: string = this.height || '100%';
+    allowfullscreen: boolean = this.allowfullscreen || true;
+    version: string | null = this.version || null;
+    isLoading: boolean = this.isLoading || true;
+
+    @computed('download', 'version')
+    get mfrUrl(this: FileRenderer) {
+        let download = `${this.get('download')}?direct&mode=render&initialWidth=766`;
+        if (this.get('version')) {
+            download += `&version=${this.get('version')}`;
```

Comment:
Is TS smart enough to know that `this.get('version')` isn't null here? Or does that only work for normal variables?

Reply:
It appears to not. Even though it does know that `this.get('version')` is `string | null`, the type guard doesn't seem to guarantee that it's just `string`. Maybe because the `.get()` could potentially return something different on the second call?